src/System.Management.Automation/engine/Modules/ModuleUtils.cs

```diff
@@ -146,59 +168,75 @@ internal static IEnumerable<string> GetDefaultAvailableModuleFiles(bool force, b
             }
         }
 
-        internal static List<string> GetModuleVersionsFromAbsolutePath(string directory)
+        /// <summary>
+        /// Get a list of module files from the given directory without recursively searching all sub-directories.
+        /// This method assumes the given directory is a module folder or a version sub-directory of a module folder.
+        /// </summary>
+        internal static List<string> GetModuleFilesFromAbsolutePath(string directory)
         {
             List<string> result = new List<string>();
             string fileName = Path.GetFileName(directory);
-            Version moduleVersion;
-            // if the user give the module path including version, we should be able to find the module as well
-            if (Version.TryParse(fileName, out moduleVersion) && Directory.Exists(Directory.GetParent(directory).ToString()))
-            {
-                fileName = Directory.GetParent(directory).Name;
-            }
-            foreach (var version in GetModuleVersionSubfolders(directory))
+
+            // If the given directory doesn't exist or it's the root folder, then return an empty list.
+            if (!Directory.Exists(directory) || fileName == string.Empty) { return result; }
```

Comment:
MSDN [suggests using `String.IsNullOrEmpty()`](https://msdn.microsoft.com/en-us/library/system.string.empty(v=vs.110).aspx#Anchor_1) instead of `== String.Empty` here. Does a `null` filename mean something else?

Reply:
Also, it might be worth moving these checks up above where we allocate the list

Reply:
I agree, `NullOrEmpty()` is the better practice, makes it easier to read an less error prone since NullReferenceExceptions are the most expensive unintentional exceptions in this world.

Reply:
The `GetModuleFilesFromAbsolutePath` method is removed along with the refactor to `GetModuleForRootedPaths`. It's not needed for the proposed new behaviors.